processmanager/ebpf/ebpf.go

```diff
@@ -253,29 +253,42 @@ func LoadMaps(ctx context.Context, maps map[string]*cebpf.Map) (EbpfHandler, err
 // UpdateInterpreterOffsets adds the given moduleRanges to the eBPF map interpreterOffsets.
 func (impl *ebpfMapsImpl) UpdateInterpreterOffsets(ebpfProgIndex uint16, fileID host.FileID,
 	offsetRanges []util.Range) error {
-	if offsetRanges == nil {
-		return errors.New("offsetRanges is nil")
-	}
-	for _, offsetRange := range offsetRanges {
-		//  The keys of this map are executable-id-and-offset-into-text entries, and
-		//  the offset_range associated with them gives the precise area in that page
-		//  where the main interpreter loop is located. This is required to unwind
-		//  nicely from native code into interpreted code.
-		key := uint64(fileID)
-		value := C.OffsetRange{
-			lower_offset:  C.u64(offsetRange.Start),
-			upper_offset:  C.u64(offsetRange.End),
-			program_index: C.u16(ebpfProgIndex),
-		}
-		if err := impl.interpreterOffsets.Update(unsafe.Pointer(&key), unsafe.Pointer(&value),
-			cebpf.UpdateAny); err != nil {
-			log.Fatalf("Failed to place interpreter range in map: %v", err)
-		}
+	key, value, err := InterpreterOffsetKeyValue(ebpfProgIndex, fileID, offsetRanges)
+	if err != nil {
+		return err
+	}
+	if err := impl.interpreterOffsets.Update(unsafe.Pointer(&key), unsafe.Pointer(&value),
+		cebpf.UpdateAny); err != nil {
+		log.Fatalf("Failed to place interpreter range in map: %v", err)
 	}
 
 	return nil
 }
 
+func InterpreterOffsetKeyValue(ebpfProgIndex uint16, fileID host.FileID,
+	offsetRanges []util.Range) (key uint64, value C.OffsetRange, err error) {
+	if len(offsetRanges) != 1 && len(offsetRanges) != 2 {
+		return 0, C.OffsetRange{}, fmt.Errorf("ivalid ranges %+v", offsetRanges)
```

Comment:
Typo and do we need `%+v` here?

```suggestion
		return 0, C.OffsetRange{}, fmt.Errorf("invalid ranges %v", offsetRanges)
```